src/test/java/seedu/address/AppParametersTest.java

```diff
@@ -17,11 +18,20 @@ public class AppParametersTest {
     private final ParametersStub parametersStub = new ParametersStub();
     private final AppParameters expected = new AppParameters();
 
+    @Test
+    public void equals() {
+        assertEquals(expected, expected);
+        // Tests for incompatible types for code coverage in equals method
+        assertNotEquals(expected, 1);
+    }
+
     @Test
     public void parse_validConfigPath_success() {
         parametersStub.namedParameters.put("config", "config.json");
         expected.setConfigPath(Paths.get("config.json"));
-        assertEquals(expected, AppParameters.parse(parametersStub));
+        AppParameters test = AppParameters.parse(parametersStub);
+        assertEquals(expected, test);
+        assertEquals(expected.hashCode(), test.hashCode());
```

Comment:
Line 34 is redundant here. That is because Line 33 implies that expected == test. But this also implies their hashes are equal too.

Reply:
This is to increase code coverage for the hashcode method.

Reply:
Ok understand!